resolver/resolver.cc

```diff
@@ -1362,7 +1356,8 @@ class ResolveConstantsWalk {
             const bool possibleGenericType = false;
             if (resolveConstantJob(ctx, nesting_, constant, resolutionFailed, possibleGenericType)) {
                 categoryCounterInc("resolve.constants.nonancestor", "firstpass");
-                if (loadScopeDepth_ == 0) {
+                if (loadScopeDepth_ == 0 && (!constant->symbol.isClassOrModule() ||
+                                             constant->symbol.asClassOrModuleRef().data(ctx)->isDeclared())) {
```

Comment:
Do you think it's worth briefly mentioning why we only report reference order for declared comments?

> While Sorbet treats `class A::B; end` like an implicit definition of `A`, it's actually a
> reference of `A`--Ruby will require a proper definition of `A` elsewhere. Long term,
> Sorbet should be taught to emit errors when these references are not actually defined,
> matching Ruby's behavior. Then the reference order checks will be able to check all
> references against their definitions, and not limit them to only isDeclared symbols here.
>
> (Historically, Stripe's custom autoloader used static analysis to predeclare these
> intermediate namespaces, so they would always be defined at the right time. As Stripe's
> codebase moves away from this legacy autoloader, it will be easier to introduce such
> changes into Sorbet.)